src/main/java/slimeknights/tconstruct/gadgets/entity/shuriken/RenderShuriken.java

```diff
@@ -0,0 +1,45 @@
+package slimeknights.tconstruct.gadgets.entity.shuriken;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import net.minecraft.client.renderer.IRenderTypeBuffer;
+import net.minecraft.client.renderer.ItemRenderer;
+import net.minecraft.client.renderer.entity.EntityRendererManager;
+import net.minecraft.client.renderer.entity.SpriteRenderer;
+import net.minecraft.util.math.vector.Quaternion;
+
+public class RenderShuriken extends SpriteRenderer<ShurikenEntityBase> {
+
+  public RenderShuriken(EntityRendererManager renderManagerIn, ItemRenderer itemRendererIn) {
+    super(renderManagerIn, itemRendererIn);
+  }
+
+  @Override
+  public void render(ShurikenEntityBase entity, float entityYaw, float partialTicks, MatrixStack matrixStackIn, IRenderTypeBuffer bufferIn, int packedLightIn) {
+    matrixStackIn.push();
+    matrixStackIn.scale(0.6F, 0.6F, 0.6F);
+
+    // TODO Figure out rotation stuff if the current isn't good enough
+    // original credits to Boni
+    matrixStackIn.rotate(new Quaternion(1.0F, entity.rotationYaw, 0F, true));
+    matrixStackIn.rotate(new Quaternion(-entity.rotationPitch, 1.0F, 0F, true));
+
+    // add some diversity
+    matrixStackIn.rotate(new Quaternion(entity.rotationPitch, 0F, 1.0F, true));
+
+    // rotate it into a horizontal position
+    matrixStackIn.rotate(new Quaternion(90.0F, 0.0F, 0.0F, true));
```

Comment:
Something is off with the rotation, I get a vertical sideview when throwing it east or west and a horizontal sideview when throwing it north or south. Look at the base class for sprite renderer, maybe you can just undo those rotations by applying the opposite. Or reimplement that class instead of extending it to replace the rotations

Reply:
Couldn't seem to figure this one out, I've never worked with Quaternions. If you could take a look, that would be great.